### DISCUSSION TITLE
Is there an output bit rate limit for VVenc?

### DISCUSSION THREAD

**Discussion by SangjinHahm** [fraunhoferhhi/vvenc#190] [Q&A] [answered]
I am testing VVC performance in the content production environment with VVenc.
Therefore, VVC performance using only I frame is being tested by comparing it with other studio codecs.
The I frame-only option is being tested using the method proposed here in Discussion.
--IntraPeriod=1 --GOPSize=1 --MaxParallelFrames=1

The problem is that HD video (422, 8 bits, 25 fps) does not produce a bit rate output greater than 250 Mbps for high quality testing.
The parameters of the VVenc experiment I used are as follows.
vvencFFapp -i tractor_1080p25.y4m --IntraPeriod=1 --GOPSize=1 --MaxParallelFrames=1 --NumPasses 2  --TargetBitrate 220M  --MSEBasedSequencePSNR 0 --PrintHexPSNR 1 -b tractor_HD_220M_VVC.266

The output stream did not come out at 220 Mbps and came out at about 140 Mbps.
Is the coding parameter I used wrong? Or, I want to know if there is a maximum output bit rate of VVenc.

**Comment by adamjw24** [comment, marked as answer]
We do have some problems with RC and high bit-rates, so this might be an related issue, alas it is more present in single pass encoding. Some of those problems have been resolved in https://github.com/fraunhoferhhi/vvenc/tags/v1.6.0-rc1. Additionally, our RC is mostly customized for random-access coding, the all-intra (AI) scenario has not been tested for a long time.

As a workaround, you might try setting your `TargetBitrate` higher or using fix-QP with a `QP` of 10 or similar, but this would require some trial-and-error until you can reach the desired rate.

For now I am afraid, I cannot give you a satisfying answer. That's an application outside of the main scope of VVenCs functionalities, thus you are using it at your own risk for now. We might have a look at some point, especially into RC+AI, but the application you described might never become a first-order priority.
